lib/semian/resource.rb

```diff
@@ -5,7 +5,7 @@ class Resource #:nodoc:
     class << Semian::Resource
       # Ensure that there can only be one resource of a given type
       def instance(*args)
-        Semian.resources[args.first] ||= new(*args)
+        Semian.resources[args.first] ||= ProtectedResource.new(args.first, new(*args), nil)
```

Comment:
https://github.com/Shopify/semian/blob/master/test/resource_test.rb#L80 was failing since we were using a `Resource` with `Semain.resources`. The error was raising [here](https://github.com/Shopify/semian/pull/193/files#diff-0c0b1885ac3076e71428d7414e517530R51) since `circuit_breaker` does not exist with `Resource`. Let me know if this fix make sense.

Reply:
Yeah, this seems correct. I think that "Resource" and "Bulkhead" are somewhat convoluted here. A useful refactor would be to change `Semian::Resource` to `Semian::Bulkhead` instead. I think it's sensible to change a resource to always be a "protected resource"; with it having a bulkhead and circuit object within.

Reply:
I want to emphasize that I find this quite confusing.
This should either be made less confusing or a comment should be added IMO.

Reply:
I would prefer to make it less confusing without having to add a comment. Do you have any idea how I could do that? I honestly don't really know how to make it simpler without modifying any behaviour.